conan/tools/google/bazeldeps.py

```diff
@@ -1,278 +1,588 @@
+    return ['"{}/**"'.format(_relativize_path(path, package_folder_path))
+            for path in cpp_info.includedirs]
+
+
+def _get_includes(cpp_info, package_folder_path):
+    return ['"{}"'.format(_relativize_path(path, package_folder_path))
+            for path in cpp_info.includedirs]
+
+
+def _get_defines(cpp_info):
+    return ['"{}"'.format(define.replace('"', '\\' * 3 + '"'))
+            for define in cpp_info.defines]
+
+
+def _get_linkopts(cpp_info, os_build):
+    link_opt = '/DEFAULTLIB:{}' if os_build == "Windows" else '-l{}'
+    system_libs = [link_opt.format(lib) for lib in cpp_info.system_libs]
+    shared_flags = cpp_info.sharedlinkflags + cpp_info.exelinkflags
+    return [f'"{flag}"' for flag in (system_libs + shared_flags)]
+
+
+def _get_copts(cpp_info):
+    # FIXME: long discussions between copts (-Iflag) vs includes in Bazel. Not sure yet
+    # includedirsflags = ['"-I{}"'.format(_relativize_path(d, package_folder_path))
+    #                     for d in cpp_info.includedirs]
+    cxxflags = [var.replace('"', '\\"') for var in cpp_info.cxxflags]
+    cflags = [var.replace('"', '\\"') for var in cpp_info.cflags]
+    return [f'"{flag}"' for flag in (cxxflags + cflags)]
+
+
+def _relativize_path(path, pattern):
+    """
+    Returns a relative path with regard to pattern given.
+
+    :param path: absolute or relative path
+    :param pattern: either a piece of path or a pattern to match the leading part of the path
+    :return: Unix-like path relative if matches to the given pattern.
+             Otherwise, it returns the original path.
+    """
+    if not path or not pattern:
+        return path
+    path_ = path.replace("\\", "/").replace("/./", "/")
+    pattern_ = pattern.replace("\\", "/").replace("/./", "/")
+    match = re.match(pattern_, path_)
+    if match:
+        matching = match[0]
+        if path_.startswith(matching):
+            path_ = path_.replace(matching, "").strip("/")
+            return path_.strip("./") or "./"
+    return path
+
+
+class _BazelDependenciesBZLGenerator:
+    """
+    Bazel needs to know all the dependencies for its current project. So, the only way
+    to do that is to tell the WORKSPACE file how to load all the Conan ones. This is the goal
+    of the function created by this class, the ``load_conan_dependencies`` one.
+
+    More information:
+        * https://bazel.build/reference/be/workspace#new_local_repository
+    """
+
+    filename = "dependencies.bzl"
+    template = textwrap.dedent("""\
+        # This Bazel module should be loaded by your WORKSPACE file.
+        # Add these lines to your WORKSPACE one (assuming that you're using the "bazel_layout"):
+        # load("@//conan:dependencies.bzl", "load_conan_dependencies")
+        # load_conan_dependencies()
+
+        {% macro new_local_repository(pkg_name, pkg_folder, pkg_build_file_path) %}
+            native.new_local_repository(
+                name="{{pkg_name}}",
```

Comment:
I think this should take the name from `bazel_repository_name` but apparently it does not, also is the property properly covered by tests as the tests for the property were not failing?